frontend/src/components/molecules/Modal.tsx

```diff
@@ -1,15 +1,49 @@
 import React from "react";
+import { Box } from "@mui/material";
+import { Typography } from "@mui/material";
+import { Grid } from "@mui/material";
+import Button from "../../components/atoms/Button";
 
 interface ModalProps {
   open: boolean;
   body: React.ReactElement;
+  title: string;
+  text: string;
 }
 
 /**
  * A Modal component is a styled modal that takes in a body component
  */
 const Modal = ({ body }: ModalProps) => {
-  return <>Hello there</>;
+  return (
+    <>
+      <Box
+        padding={10}
+        sx={{
+          width: 500,
+          height: 100,
+          backgroundColor: "white",
+          alignItems: "center",
+          border: 1,
+        }}
+      >
+        <Typography id="modal-modal-title" variant="h6" component="h2">
+          Text in a modal
+        </Typography>
+        <Typography id="modal-modal-description" sx={{ mt: 2 }}>
+          Description of the modal
+        </Typography>
```

Comment:
Kindly use https://mui.com/material-ui/react-modal/. It is hard to make a modal from scratch using tailwind